src/SlewLimiter.cpp

```diff
@@ -19,49 +21,74 @@ struct SlewLimiter : Module {
 		NUM_OUTPUTS
 	};
 
-	float out = 0.0;
+	float_4 out[4];
 
 	SlewLimiter() {
 		config(NUM_PARAMS, NUM_INPUTS, NUM_OUTPUTS);
 		configParam(SHAPE_PARAM, 0.0, 1.0, 0.0, "Shape");
 		configParam(RISE_PARAM, 0.0, 1.0, 0.0, "Rise time");
 		configParam(FALL_PARAM, 0.0, 1.0, 0.0, "Fall time");
+
+		memset(out, 0, sizeof(out));
 	}
 
-	void process(const ProcessArgs &args) override {
-		float in = inputs[IN_INPUT].getVoltage();
-		float shape = params[SHAPE_PARAM].getValue();
+	void process(const ProcessArgs& args) override {
+
+		float_4 in[4] = {0.f};
+		float_4 riseCV[4] = {0.f};
+		float_4 fallCV[4] = {0.f};
 
-		// minimum and maximum slopes in volts per second
+		// this is the number of active polyphony engines, defined by the input
+		int numPolyphonyEngines = inputs[IN_INPUT].getChannels();
+
+		// minimum and std::maximum slopes in volts per second
 		const float slewMin = 0.1;
 		const float slewMax = 10000.f;
 		// Amount of extra slew per voltage difference
-		const float shapeScale = 1/10.f;
-
-		// Rise
-		if (in > out) {
-			float rise = inputs[RISE_INPUT].getVoltage() / 10.f + params[RISE_PARAM].getValue();
-			float slew = slewMax * std::pow(slewMin / slewMax, rise);
-			out += slew * crossfade(1.f, shapeScale * (in - out), shape) * args.sampleTime;
-			if (out > in)
-				out = in;
-		}
-		// Fall
-		else if (in < out) {
-			float fall = inputs[FALL_INPUT].getVoltage() / 10.f + params[FALL_PARAM].getValue();
-			float slew = slewMax * std::pow(slewMin / slewMax, fall);
-			out -= slew * crossfade(1.f, shapeScale * (out - in), shape) * args.sampleTime;
-			if (out < in)
-				out = in;
-		}
+		const float shapeScale = 1 / 10.f;
+
+		const float_4 shape = float_4(params[SHAPE_PARAM].getValue());
+		const float_4 param_rise = float_4(params[RISE_PARAM].getValue() * 10.f);
+		const float_4 param_fall = float_4(params[FALL_PARAM].getValue() * 10.f);
+
+		outputs[OUT_OUTPUT].setChannels(numPolyphonyEngines);
+
+		for (int c = 0; c < numPolyphonyEngines; c += 4) {
+			in[c / 4] = inputs[IN_INPUT].getVoltageSimd<float_4>(c);
 
-		outputs[OUT_OUTPUT].setVoltage(out);
+			if (inputs[RISE_INPUT].isConnected()) {
+				riseCV[c / 4] = inputs[RISE_INPUT].getPolyVoltageSimd<float_4>(c);
+			}
+			if (inputs[FALL_INPUT].isConnected()) {
+				fallCV[c / 4] = inputs[FALL_INPUT].getPolyVoltageSimd<float_4>(c);
+			}
+
+			riseCV[c / 4] += param_rise;
+			fallCV[c / 4] += param_fall;
+
+			float_4 delta = in[c / 4] - out[c / 4];
+			float_4 delta_gt_0 = delta > float_4::zero();
```

Comment:
I have doubts that `float_4::zero()` is any faster than `simd::Vector(float)`, so you can replace that with just `0.f` everywhere in this PR.

Reply:
Good to know, done.